controller/controller_test.go

```diff
@@ -498,7 +500,8 @@ func TestStartAndShutdownWithErroringWork(t *testing.T) {
 
 	var eg errgroup.Group
 	eg.Go(func() error {
-		return impl.Run(1, stopCh)
+		StartAll(stopCh, impl)
```

Comment:
this (and others) can be redone with channels as the new tests, so sleeps can go away.

Reply:
Done, but it ends up being longer.

Reply:
But no sleeps! :)